opm/simulators/wells/StandardWell_impl.hpp

```diff
@@ -868,19 +868,19 @@ namespace Opm
     void
     StandardWell<TypeTag>::
     updatePrimaryVariablesNewton(const BVectorWell& dwells,
-                                 const WellState& /* well_state */) const
+                                 const WellState& /* well_state */,
+                                 DeferredLogger& deferred_logger) const
     {
         const double dFLimit = this->param_.dwell_fraction_max_;
         const double dBHPLimit = this->param_.dbhp_max_rel_;
         this->StdWellEval::updatePrimaryVariablesNewton(dwells, dFLimit, dBHPLimit);
 
         updateExtraPrimaryVariables(dwells);
 
-#ifndef NDEBUG
         for (double v : this->primary_variables_) {
-            assert(isfinite(v));
+            if(!isfinite(v))
+                OPM_DEFLOG_THROW(NumericalIssue, "Infinite primary variable after newton update well: " << this->name(),  deferred_logger);
```

Comment:
Just to be sure this is what you actually want: you will now (as with the existing assert()) always, unconditionally, react to the condition, but the reaction will not be terminating the program but throwing an exception. (Which will be picked up higher up in the code, communicated properly,  and result in a timestep cut.)

It sounds good to me, so if I understood you correctly, this can be merged. (Although I would appreciate adding spaces after the `if`s and braces around the bodies of the `if`s. We write macros that are safe for this, but not all do.)